Feature/EL-U2 Cambios de User agregados

Victoriasaurio opened the pull request:
Se añadió:
- Paginación (offset con 0 por defecto, limit con 20 por defecto). También soporta las query's que el usuario mande, como el rol y active, al igual que permite modificar los valores de offset y limit para devolver la lista deseada.
- Las nuevas funciones agregadas están comentadas.
- Contiene test de rutas con Jest y Supertest al igual que documentación con swagger.

Resuelve issue: #8

Commit:
Merge pull request #1 from alvarez98/develop

Agregando estructura y db

Commit:
Update README.md

Victoriasaurio committed:
Merge branch 'develop' of https://github.com/alvarez98/e-learning-system into develop

Victoriasaurio committed:
Merge branch 'develop' of https://github.com/alvarez98/e-learning-system into develop

Victoriasaurio committed:
Restart
- Se le añadió la propiedad restart a la imagen postgres:10-alphine

Victoriasaurio committed:
getUser
- Función getUser añadida.
- Contiene la función buildUserFilters, que se encarga de almacenar las querys predeterminadas y las otorgadas por el usuario
- Contiene la función getUserOnDB, encargada de buscar en la base de datos, la lista señalada por medio de las querys.
- Señala la lista total de objetos existentes, así como también el total de objetos desplegados dependiendo del offset y limit.

Victoriasaurio committed:
getUserOnDB
- Función que devuelve la lista de usuarios dependiendo de los filsters(rol, usuario activo o no), en caso de no haber filters específicos devuelve una lista de todos los usuario existentes.
-

Victoriasaurio committed:
Ruta get users añadida, junto con sus respectivas validaciones

Victoriasaurio committed:
UserGetSchema
- Esquema que valida las querys (active, limit, offset y rol)

Victoriasaurio committed:
Agregando documentación
- Petición GET añadida con la documentación Swagger

Victoriasaurio committed:
Test
- Prueba test añadida, que devuelve un status 200, cuando la petición a sido exitosa.
- 3 pruebas test añadidas, de posibles errores.

Victoriasaurio committed:
buildUserFilters
- Función que recibe las query's del usuario.
- Función que lee las query's predeterminadas.
- Guarda las peticiones query's en un objeto llamado filters. Nota: Las query's debe ser peticiones válidas.
- Retorna un objeto que almacena: filters(active y rol), offset y limit.

Alvarez-P commented on src/tests/user.test.js:
```diff
@@ -2,20 +2,64 @@ const request = require("supertest");
+    });
+});
+
+describe("GET /users", () => {
```
Los 3 test van dentro del mismo describe

Alvarez-P replied:
Los test que sean gets, van dentro del mismo describe("GET /users ", () => {
    test("", done =>{
    })
    test("", done =>{
    })
    test("", done =>{
    })
}

Alvarez-P reviewed (changes requested):
Corregir los tests

Victoriasaurio committed:
Archivo test corregido
- Test unidos en un solo describe.
- Códigos de error utilizados en cada caso del test según corresponda.
